src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/bufferOutputPolling.ts

```diff
@@ -215,3 +240,37 @@ export async function assessOutputForErrors(buffer: string, token: CancellationT
 		return 'Error occurred ' + err;
 	}
 }
+
+/**
+ * Returns true if the last line of the output is a prompt asking for user input (e.g., "Do you want to continue? (y/n)").
+ * This is a heuristic and matches common prompt patterns.
+ */
+/**
+ * Returns the kind of input expected by the last line of the output if it appears to be a user prompt.
+ * For example, returns 'y/n', 'yes/no', 'press key', 'choice', etc., or undefined if not a prompt.
+ */
+export function getExpectedUserInputKind(output: string): string | undefined {
+	if (!output) {
+		return undefined;
+	}
+
+	const patterns: { regex: RegExp; kind: string }[] = [
+		{ regex: /\bdo you want to continue\b.*\(y\/n\)/ig, kind: 'y/n' },
+		{ regex: /\bcontinue\? \(y\/n\)/ig, kind: 'y/n' },
+		{ regex: /\bproceed\? \(y\/n\)/ig, kind: 'y/n' },
+		{ regex: /\btype yes or no\b/ig, kind: 'yes/no' },
+		{ regex: /\b\(yes\/no\)/ig, kind: 'yes/no' },
+		{ regex: /\b\[y\/n\]/ig, kind: 'y/n' },
+		{ regex: /\benter your choice\b/ig, kind: 'choice' },
+		{ regex: /\bselect an option\b/ig, kind: 'choice' },
+		{ regex: /\bplease respond\b/ig, kind: 'response' },
+		{ regex: /\bpress (enter|return|any key)\b/ig, kind: 'press key' }
+	];
```

Comment:
Did you have programs that output these? We should include the programs in comments if so, so we know the source.